src/main/java/duke/command/FindCommand.java

```diff
@@ -0,0 +1,38 @@
+package duke.command;
+
+import java.util.List;
+
+import duke.task.Task;
+import duke.util.TaskList;
+import duke.util.Ui;
+
+public class FindCommand extends Command {
+
+    private String matchString;
+
+    public FindCommand(String matchString) {
+        this.matchString = matchString;
+    }
+
+    public String execute(TaskList taskList, Ui ui) {
```

Comment:
Perhaps a bit lengthy code here? Not sure if some parts could be abstracted out of the execute method......